pkg/route/api/v1/conversion.go

```diff
@@ -13,6 +13,9 @@ func init() {
 			obj.To.Kind = "Service"
 		},
 		func(obj *TLSConfig) {
+			if len(obj.Termination) == 0 {
+				obj.Termination = TLSTerminationEdge
```

Comment:
If we want to create a new type TerminationNone that would be more accurate
(and default to that).  However, a user who creates and adds certs almost
certainly wants termination, not "none"

On Mon, Dec 14, 2015 at 9:07 AM, David Eads <notifications@github.com>
wrote:

> In pkg/route/api/v1/conversion.go
> <https://github.com/openshift/origin/pull/6076#discussion_r47500560>:
>
> > @@ -13,6 +13,9 @@ func init() {
> >  			obj.To.Kind = "Service"
> >  		},
> >  		func(obj *TLSConfig) {
> > +			if len(obj.Termination) == 0 {
> > +				obj.Termination = TLSTerminationEdge
>
> What if they are set on purpose (plans for enabling https at a future
> point? Not sure how valid of a use case it is) and we silently remove them?
>
> I think I'm ok with removing them. It seems unlikely that I've chosen to
> use an openshift route as the sole location for storing my certs and keys,
> so it should be recoverable if that was their intent. It would also be a
> pretty unusual intent: "I want this to be secure, but I'll start out my
> making it insecure"?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6076/files#r47500560>.
>